lib/puppet/provider/package/yum.rb

```diff
@@ -13,6 +13,30 @@
 
   commands :cmd => "yum", :rpm => "rpm"
 
+  YUM_ARCH_LIST = [
+    'i386',
+    'i686',
+    'ppc',
+    'ppc64',
+    'armv3l',
+    'armv4b',
+    'armv4l',
+    'armv4tl',
+    'armv5tel',
+    'armv5tejl',
+    'armv6l',
+    'armv7l',
+    'm68kmint',
+    's390',
+    's390x',
+    'ia64',
+    'x86_64',
+    'sh3',
+    'sh4',
+  ]
+
+  ARCH_REGEX = Regexp.new(YUM_ARCH_LIST.join('|\.'))
```

Comment:
Also is arch always last in the package name? If so, we could tighten the regex some accordingly.